File: BroadcastReceivers/RegisteringBroadcast/app/src/main/java/com/tomerpacific/broadcastreceiver/MainActivity.java
package com.tomerpacific.broadcastreceiver;

import android.content.BroadcastReceiver;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.ConnectivityManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    private Button sendBroadcastBtn;
    private BroadcastReceiver myBroadcastReceiver = new MyBroadcastReceiver();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final IntentFilter intentFilter = new IntentFilter("MY_ACTION");
        this.registerReceiver(myBroadcastReceiver, intentFilter);

        sendBroadcastBtn = findViewById(R.id.sendBroadcastBtn);
        sendBroadcastBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent("MY_ACTION");
                intent.putExtra("data", "Hello World!");
                sendBroadcast(intent);
            }
        });
    }

    @Override
    protected void onStop() {
        super.onStop();
        this.unregisterReceiver(myBroadcastReceiver);
    }
}

File: BroadcastReceivers/RegisteringBroadcast/app/src/main/java/com/tomerpacific/broadcastreceiver/MainActivity.kt
package com.tomerpacific.broadcastreceiver

import androidx.appcompat.app.AppCompatActivity
import android.content.BroadcastReceiver
import android.os.Bundle
import android.content.IntentFilter
import android.content.Intent
import android.view.View
import android.widget.Button

class MainActivity : AppCompatActivity() {
    private val myBroadcastReceiver: BroadcastReceiver = MyBroadcastReceiver()
    private val ACTION_KEY: String = "MY_ACTION"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val intentFilter = IntentFilter(ACTION_KEY)
        this.registerReceiver(myBroadcastReceiver, intentFilter)
        findViewById<Button?>(R.id.sendBroadcastBtn).apply {
            setOnClickListener(View.OnClickListener {
                val intent = Intent(ACTION_KEY)
                intent.putExtra("data", "Hello World!")
                sendBroadcast(intent)
            })
        }

    }

    override fun onStop() {
        super.onStop()
        unregisterReceiver(myBroadcastReceiver)
    }
}